### ISSUE TITLE
clearName function and emulators on local dev environment

### ISSUE THREAD

**Issue by jpablobr** [Miodec/monkeytype#604]
I think there's a problem with `/functions/clearName`. I was only able to deploy with `firebase deploy --except functions` after the following log error message:

```
Functions deploy had errors with the following functions:
	clearName


To try redeploying those functions, run:
    firebase deploy --only "functions:clearName"


To continue deploying other features (such as database), run:
    firebase deploy --except functions

Error: Functions did not deploy properly.
```

Still, not sure if that function is needed but I can't register users:

https://monkeytype-fb21e.web.app/login

On a different but kind of related note, I was following the setup instructions in CONTRIBUTING.md but still not sure why the `npm run start:dev` only uses the `functions` emulator (`"firebase emulators:start --only functions"`) and not the `Firestore` and maybe(?) the `Authentication` emulators as well. It might be worth adding those details to the setup, as in a normal dev workflow kind of way. As it is, seems like local development is against a remote storage (Cloud Firestore) making it cumbersome and potentially costly. Based on [this discussion](https://github.com/firebase/firebase-functions/issues/4) I gather that emulators is the way to go locally. 

Edit:

I get the same error message when trying to register a user locally (localhost:5000). Maybe related to https://github.com/Miodec/monkeytype/issues/577 ?

**Comment by Miodec** [comment]
What is the exact error youre getting when trying to deploy it

**Comment by Miodec** [comment]
Also, you seem to have old code, because the `start:dev` command is `npm run build && concurrently --kill-others \"npx gulp watch\" \"firebase serve\"`

**Comment by jpablobr** [comment]
I think I got it now, had to configure the app's Authentication in the Firebase console (or at least that seems to have done it). I only had Firestore configured. That also seems to have fixed the problem with the `clearName` function.

As to the `start:dev`, you're right, not sure where I got that from but things do seem to work now (I'm able to create users locally - not with the emulators but for now that's fine).
